# Task 1 – Dataset Ingestion and Validation
**Agent:** Claude (claude-sonnet-4-6)  
**Date:** 2026-03-15  
**Dataset:** `data/processed/airline_passenger_satisfaction_full.csv`

In [1]:
import pandas as pd
import numpy as np

## 1. Data Ingestion

In [2]:
df = pd.read_csv("../../data/processed/airline_passenger_satisfaction_full.csv")

print(f"Rows:    {df.shape[0]:,}")
print(f"Columns: {df.shape[1]}")

Rows:    129,880
Columns: 24


## 2. Schema Inspection

In [3]:
print("Column names:")
for col in df.columns:
    print(f"  - {col}")

Column names:
  - id
  - Gender
  - Customer Type
  - Age
  - Type of Travel
  - Class
  - Flight Distance
  - Inflight wifi service
  - Departure/Arrival time convenient
  - Ease of Online booking
  - Gate location
  - Food and drink
  - Online boarding
  - Seat comfort
  - Inflight entertainment
  - On-board service
  - Leg room service
  - Baggage handling
  - Checkin service
  - Inflight service
  - Cleanliness
  - Departure Delay in Minutes
  - Arrival Delay in Minutes
  - satisfaction


In [4]:
print("Data types:")
print(df.dtypes.to_string())

Data types:
id                                     int64
Gender                                object
Customer Type                         object
Age                                    int64
Type of Travel                        object
Class                                 object
Flight Distance                        int64
Inflight wifi service                  int64
Departure/Arrival time convenient      int64
Ease of Online booking                 int64
Gate location                          int64
Food and drink                         int64
Online boarding                        int64
Seat comfort                           int64
Inflight entertainment                 int64
On-board service                       int64
Leg room service                       int64
Baggage handling                       int64
Checkin service                        int64
Inflight service                       int64
Cleanliness                            int64
Departure Delay in Minutes             int6

## 3. Missing Value Check

In [5]:
missing = df.isnull().sum()
missing_pct = (missing / len(df) * 100).round(4)
missing_report = pd.DataFrame({
    "Missing Count": missing,
    "Missing %": missing_pct
})
cols_with_missing = missing_report[missing_report["Missing Count"] > 0]

if len(cols_with_missing) == 0:
    print("No missing values found.")
else:
    print(f"Columns with missing values ({len(cols_with_missing)}):")
    display(cols_with_missing)
    print(f"\nTotal missing cells: {missing.sum():,}")

Columns with missing values (1):


,Missing Count,Missing %
Arrival Delay in Minutes,393,0.3026



Total missing cells: 393


## 4. Summary Statistics

In [6]:
print("Numeric columns – summary statistics:")
display(df.describe().T)

Numeric columns – summary statistics:


,count,mean,std,min,25%,50%,75%,max
id,129880.0,64940.500000,37493.270818,1.0,32470.75,64940.5,97410.25,129880.0
Age,129880.0,39.427957,15.119360,7.0,27.00,40.0,51.00,85.0
Flight Distance,129880.0,1190.316392,997.452477,31.0,414.00,844.0,1744.00,4983.0
Inflight wifi service,129880.0,2.728696,1.329340,0.0,2.00,3.0,4.00,5.0
Departure/Arrival time convenient,129880.0,3.057599,1.526741,0.0,2.00,3.0,4.00,5.0
Ease of Online booking,129880.0,2.756876,1.401740,0.0,2.00,3.0,4.00,5.0
Gate location,129880.0,2.976925,1.278520,0.0,2.00,3.0,4.00,5.0
Food and drink,129880.0,3.204774,1.329933,0.0,2.00,3.0,4.00,5.0
Online boarding,129880.0,3.252633,1.350719,0.0,2.00,3.0,4.00,5.0
Seat comfort,129880.0,3.441361,1.319289,0.0,2.00,4.0,5.00,5.0


In [7]:
print("Categorical columns – value counts:")
cat_cols = df.select_dtypes(include="object").columns
for col in cat_cols:
    print(f"\n{col}:")
    display(df[col].value_counts().to_frame())

Categorical columns – value counts:

Gender:


,count
Gender,
Female,65899
Male,63981



Customer Type:


,count
Customer Type,
Loyal Customer,106100
disloyal Customer,23780



Type of Travel:


,count
Type of Travel,
Business travel,89693
Personal Travel,40187



Class:


,count
Class,
Business,62160
Eco,58309
Eco Plus,9411



satisfaction:


,count
satisfaction,
neutral or dissatisfied,73452
satisfied,56428


## 5. Data Preview

In [8]:
display(df.head())

,id,Gender,Customer Type,Age,Type of Travel,Class,Flight Distance,Inflight wifi service,Departure/Arrival time convenient,Ease of Online booking,...,Inflight entertainment,On-board service,Leg room service,Baggage handling,Checkin service,Inflight service,Cleanliness,Departure Delay in Minutes,Arrival Delay in Minutes,satisfaction
0,70172,Male,Loyal Customer,13,Personal Travel,Eco Plus,460,3,4,3,...,5,4,3,4,4,5,5,25,18.0,neutral or dissatisfied
1,5047,Male,disloyal Customer,25,Business travel,Business,235,3,2,3,...,1,1,5,3,1,4,1,1,6.0,neutral or dissatisfied
2,110028,Female,Loyal Customer,26,Business travel,Business,1142,2,2,2,...,5,4,3,4,4,4,5,0,0.0,satisfied
3,24026,Female,Loyal Customer,25,Business travel,Business,562,2,5,5,...,2,2,5,3,1,4,2,11,9.0,neutral or dissatisfied
4,119299,Male,Loyal Customer,61,Business travel,Business,214,3,3,3,...,3,3,4,4,3,3,3,0,0.0,satisfied


## 6. Save Cleaned Dataset

Only `Arrival Delay in Minutes` has missing values (393 rows, 0.30%).  
We drop those rows and save the cleaned dataset.

In [9]:
df_clean = df.dropna(subset=["Arrival Delay in Minutes"]).reset_index(drop=True)
df_clean.to_csv("../../data/processed/Claude_clean.csv", index=False)

print(f"Original shape : {df.shape[0]:,} rows x {df.shape[1]} columns")
print(f"Clean shape    : {df_clean.shape[0]:,} rows x {df_clean.shape[1]} columns")
print(f"Rows dropped   : {df.shape[0] - df_clean.shape[0]:,}")
print("Saved to: data/processed/Claude_clean.csv")

Original shape : 129,880 rows x 24 columns
Clean shape    : 129,487 rows x 24 columns
Rows dropped   : 393
Saved to: data/processed/Claude_clean.csv
